pkg/overlay/overlay.go

```diff
@@ -61,23 +66,31 @@ func generateOverlayStructure(containerDir string, rootUID, rootGID int) (string
 	return containerDir, nil
 }
 
+// MountWithOptions creates a subdir of the contentDir based on the source directory
+// from the source system.  It then mounts up the source directory on to the
+// generated mount point and returns the mount point to the caller.
+// But allows api to set custom workdir, upperdir and other overlay options
```

Comment:
FWIW – this is preexisting, and maybe I’m just ignorant. This documentation, even with the context of the overlay section of mount(8), is completely unintelligible to me. I have no idea how `source`/`contentDir` relate.

(Reading the code, apparently `contentDir` has a specific structure that’s not documented in this package.)

The other options are also undocumented.

Reply:
… and the behavior of `readonly`, affecting whether `$contentDir/lower` is used, is completely non-obvious to me as a newcomer to the code.